src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,82 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_CAPACITY_FACTOR = 2;
+    private int size;
+    private int capacity = DEFAULT_CAPACITY;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (capacity * DEFAULT_LOAD_FACTOR <= size) {
+            resize();
+        }
+        int index = key == null ? 0 : Math.abs(key.hashCode() % capacity);
+        Node<K,V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(key, value, null);
+                ++size;
+                return;
+            }
+            node = node.next;
+        }
+        table[index] = new Node<>(key, value, null);
+        ++size;
```

Comment:
```suggestion
            node = node.next;
        }
        node = new Node<>(key, value, null);
        table[index] = new Node<>(key, value, null);
        ++size;
```

Reply:
not fixed, this method can be simplified

Reply:
> not fixed, this method can be simplified

I have to put new node into node.next, not into node (because prev node would not have a link to next one)
and if I put new node into table[index] it will destroy all linked list hierarchy under the index